code/datums/components/simple_access.dm

```diff
@@ -0,0 +1,31 @@
+///This component allows us to give a mob access without giving them an ID card.
+/datum/component/simple_access
+	///list of accesses we are allowed to access via this component
+	var/list/access
+
+/datum/component/simple_access/Initialize(list/new_access, atom/donor_atom)
+	. = ..()
+	if(!ismob(parent))
+		return COMPONENT_INCOMPATIBLE
+	access = new_access
+	RegisterSignal(parent, COMSIG_MOB_TRIED_ACCESS, .proc/on_tried_access)
+	if(!donor_atom)
+		return
+	if(istype(donor_atom, /obj/item/organ))
+		RegisterSignal(donor_atom, COMSIG_ORGAN_REMOVED, .proc/on_donor_removed)
+	else if(istype(donor_atom, /obj/item/implant))
+		RegisterSignal(donor_atom, COMSIG_IMPLANT_REMOVED, .proc/on_donor_removed)
```

Comment:
Something is wrong here. There're signals for when the organ/implant is removed but not for when it's added. Also nowhere in the code the component is added to either of them.

Reply:
The purpose for listening to these signal is that so if the access is granted by a donor item such as an organ and implant, the component deletes itself when that donor item is removed from the mob.